pkg/monitor/cluster/hiveregistrationstatus.go

```diff
@@ -0,0 +1,50 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/Azure/ARO-RP/pkg/hive"
+)
+
+var clusterDeploymentConditionsExpected = map[hivev1.ClusterDeploymentConditionType]corev1.ConditionStatus{
+	hivev1.ClusterReadyCondition:  corev1.ConditionTrue,
+	hivev1.UnreachableCondition:   corev1.ConditionFalse,
+	hivev1.SyncSetFailedCondition: corev1.ConditionFalse,
+}
+
+func (mon *Monitor) emitHiveRegistrationStatus(ctx context.Context) error {
+	if mon.hiveclientset == nil {
+		// TODO(hive): remove this if once we have Hive everywhere
+		mon.log.Info("skipping: no hive cluster manager")
+		return nil
+	}
+
+	if mon.oc.Properties.HiveProfile.Namespace == "" {
+		return fmt.Errorf("cluster %s not adopted. No namespace in the clusterdocument", mon.oc.Name)
+	}
+
+	cd, err := mon.hiveclientset.HiveV1().ClusterDeployments(mon.oc.Properties.HiveProfile.Namespace).Get(ctx, hive.ClusterDeploymentName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, condition := range cd.Status.Conditions {
+		if expectedState, ok := clusterDeploymentConditionsExpected[condition.Type]; ok {
+			if condition.Status != expectedState {
+				mon.emitGauge("hive.conditions", 1, map[string]string{
+					"type":   string(condition.Type),
+					"reason": condition.Reason,
```

Comment:
I'm not sure about `condition.Reason` here: there are probably too many reasons/variations.

We might want to consider replacing it with `condition.Status` (it is either `True` or `False` or `Unknown`) or droping altogether.

Reply:
From what I can see the state alone is does not provide enough information for monitoring logic. For example: at adoption time the Ready condition goes false and then goes through several sub statuses (indicated by the reason), while hive is checking if VMs and COs are up, before it goes to true eventually. The combination reason and state would be required to determine if we had a real issue with the adoption of a cluster or if it's 'just' hive making up its mind. I'd like to keep that data, if possible.